packages/toolkit/src/view/settings/integrations/ConnectionForm.tsx

```diff
@@ -0,0 +1,157 @@
+import * as React from "react";
+import {
+  GeneralRecord,
+  InstillJSONSchema,
+  IntegrationMethod,
+} from "instill-sdk";
+import { z } from "zod";
+
+import { Button, cn, Form, Input } from "@instill-ai/design-system";
+
+import { LoadingSpin } from "../../../components";
+import { useInstillForm } from "../../../lib";
+import { FieldDescriptionTooltip } from "../../../lib/use-instill-form/components/common";
+
+export const connectionFormID = "connection-form";
+
+export const ConnectionForm = ({
+  id,
+  onSubmit,
+  className,
+  isProcessing,
+  additionalCta,
+  schema,
+  method,
+  values,
+  isEdit,
+}: {
+  id: string;
+  onSubmit: (props: {
+    method: IntegrationMethod;
+    payload: Record<string, unknown>;
+    id: string;
+  }) => Promise<void>;
+  className?: string;
+  isProcessing: boolean;
+  additionalCta?: React.ReactNode;
+  schema?: InstillJSONSchema;
+  method: IntegrationMethod;
+  values?: GeneralRecord;
+  isEdit?: boolean;
+}) => {
+  const { fields, form, ValidatorSchema } = useInstillForm(
+    schema || null,
+    values || null,
+  );
+  const [connectionId, setConnectionId] = React.useState("");
+
+  React.useEffect(() => {
+    if (values?.id) {
+      setConnectionId(values.id);
+    }
+  }, [values]);
+
+  const onFormSubmit = (payload: Record<string, unknown>) => {
+    const id = connectionId.trim();
+
+    if (!id) {
+      form.setError("id", {
+        message: "The component ID is required to create a connection!",
+      });
+
+      return;
+    }
+
+    if (!/^[a-z_][a-z0-9_-]{0,31}$/.test(id)) {
+      form.setError("id", {
+        message:
+          "The ID must start with a lowercase letter or underscore, followed by lowercase letters, numbers, hyphens, or underscores, and have the maximum length of 31.",
+      });
+
+      return;
+    }
+
+    form.clearErrors();
+
+    let parsedData;
+
+    try {
+      parsedData = ValidatorSchema.parse(payload);
+    } catch (err) {
+      if (err instanceof z.ZodError) {
+        console.log(err);
+      }
+
+      return;
```

Comment:
@orangecoloured We need to do something about this error since it will block user action.

Reply:
I added a toast with the error message. It's not blocking the UI. You can close the modal.